### DISCUSSION TITLE
sparse buffer

### DISCUSSION THREAD

**Discussion by hejian01** [vsg-dev/VulkanSceneGraph#973] [Q&A]
Hello, does VSG currently consider the use of sparse buffers? I've checked the source code and it seems that it doesn't support it yet. Is that correct?

**Comment by robertosfield** [comment]
I haven't attempted to tackle spare buffers/images.  My work on the VSG is driven by concrete usages cases so if I or members of the community haven't required a feature then we don't have anything to design towards, this helps keep the feature set focused on the needs of users and ensures we get the most critical features implemented first.

As time goes on the VSG is used in a wider and wider range of applications and domains so new usage cases are popping up and stretching the VSG beyond it's current capabilities.  In these cases we just need to start discussing topics to see what might be possible and how to tackle it.  Members of the community sometimes take the lead on this, other times my clients or I step up.

In the case of spare buffers/images, I haven't personally used them before so won't be able to take the lead on it, so would appreciate members of the community stepping up to chip in their experiences and suggestions.  Having a concrete usage case to test out and illustrate a feature can be useful even before the underlying VSG supports it, this could be a third party example or example built upon the VSG, tweaking/extending it to implement the required feature.
